src/software/thunderscope/thunderscope_main.py

```diff
@@ -350,16 +348,28 @@
                     robot_communication.toggle_keyboard_estop
                 )
 
-            if args.run_diagnostics:
-                for tab in tscope_config.tabs:
-                    if hasattr(tab, "widgets"):
-                        robot_view_widget = tab.find_widget("Robot View")
-                        if robot_view_widget is not None:
-                            robot_view_widget.individual_robot_control_mode_signal.connect(
-                                lambda robot_id,
-                                robot_mode: robot_communication.toggle_individual_robot_control_mode(
-                                    robot_id, robot_mode
-                                )
+            # Connect robot control mode combo box in Robot View widget to RobotCommunication
+            for tab in tscope_config.tabs:
+                if hasattr(tab, "widgets"):
+                    robot_view_widget = tab.find_widget("Robot View")
+                    if robot_view_widget is not None:
```

Comment:
I know this is unrelated but to me, this seems like a brittle way to get the robot view widget.

this should at least be refactored into a helper in `thunderscope_config.py` and we could create an issue for cleaning up this logic